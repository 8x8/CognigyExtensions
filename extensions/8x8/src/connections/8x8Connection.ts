import type { IConnectionSchema } from '@cognigy/extension-tools';

export const connection: IConnectionSchema = {
  type: 'eightbyeight',
  label: '8x8',
  fields: [
    { fieldName: 'tenantId' },
    { fieldName: 'crmApiUsername' },
    { fieldName: 'crmApiPassword' },
    { fieldName: 'dataRequestToken' },
    { fieldName: 'actionRequestToken' },
    { fieldName: 'clusterBaseUrl' }
  ]
};

export interface I8x8Connection {
  tenantId: string
  crmApiUsername: string
  crmApiPassword: string
  dataRequestToken: string
  actionRequestToken: string
  clusterBaseUrl: string
}
