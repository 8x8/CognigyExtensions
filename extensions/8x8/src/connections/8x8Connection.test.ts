import { connection } from './8x8Connection';

describe('8x8Connection', () => {
  it('should match snapshot', () => {
    expect(connection).toMatchSnapshot();
  });
});
