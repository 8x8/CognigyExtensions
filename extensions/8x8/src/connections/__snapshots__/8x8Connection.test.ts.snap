// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`8x8Connection should match snapshot 1`] = `
{
  "fields": [
    {
      "fieldName": "tenantId",
    },
    {
      "fieldName": "crmApiUsername",
    },
    {
      "fieldName": "crmApiPassword",
    },
    {
      "fieldName": "dataRequestToken",
    },
    {
      "fieldName": "actionRequestToken",
    },
    {
      "fieldName": "clusterBaseUrl",
    },
  ],
  "label": "8x8",
  "type": "eightbyeight",
}
`;
